sympy/polys/series/ring.py

```diff
@@ -1,109 +1,1007 @@
+def power_series_ring(
+    symbol: str, K: Domain, prec: int = 6
+) -> tuple[PowerSeriesRingRing | PowerSeriesRingField, PowerSeriesElement]:
     """
     Create a power series ring over the given domain.
 
     Parameters
     ==========
 
-    domain : Domain
+    symbol : str
+        The symbol to use for the power series variable.
+
+    K : Domain
         The ground domain for the power series ring. Must be ZZ or QQ.
+
     prec : int, optional
         The default precision for power series operations. Default is 6.
 
     Returns
     =======
 
-    PowerSeriesRingZZ | PowerSeriesRingQQ
-        A power series ring instance. The actual implementation (Flint or Python)
-        depends on whether python-flint is available.
+    ring : PowerSeriesRing
+        The power series ring.
+
+    generator : PowerSeriesElement
+        The generator of the power series ring.
 
     Examples
     ========
 
     >>> from sympy import QQ
     >>> from sympy.polys.series import power_series_ring
-    >>> R_QQ = power_series_ring(QQ, 5)
-    >>> inv = R_QQ.inverse(R_QQ([1, 2, (3, 4), (5, 6)]))
-    >>> R_QQ.print(inv)
-    1 - 2*x + 13/4*x**2 - 35/6*x**3 + 523/48*x**4 + O(x**5)
+    >>> R, x = power_series_ring("x", QQ)
+    >>> R.sin(x + x**2)
+    x + x**2 - 1/6*x**3 - 1/2*x**4 - 59/120*x**5 + O(x**6)
+    >>> R.log(1 + 7*x**2)
+    7*x**2 - 49/2*x**4 + O(x**6)
     """
 
     if K.is_ZZ:
-        return cast("PowerSeriesRing[Er]", PowerSeriesRingZZ(prec))
+        ring = PowerSeriesRingRing(K, symbol, prec)
+        return ring, ring.gen
     elif K.is_QQ:
-        return cast("PowerSeriesRing[Er]", PowerSeriesRingQQ(prec))
+        ring = PowerSeriesRingField(K, symbol, prec)
+        return ring, ring.gen
     else:
         raise ValueError(f"Unsupported ground domain: {K}")
+
+
+class PowerSeriesRingRing(Generic[Er]):
+    """A class for representing Univariate Power Series Rings over a Ring."""
+
+    is_PowerSeries: bool = True
+    has_assoc_Ring: bool = True
+    has_assoc_Field: bool = False
+
+    dtype: type[PowerSeriesElement[Er]]
+    domain: Domain[Er]
+    symbol: Expr
+    prec: int
+
+    def __init__(self, domain: Domain[Er], symbol: str | Expr = "x", prec: int = 6):
+        if isinstance(symbol, str):
+            symbol = Symbol(symbol)
+
+        self.dtype = PowerSeriesElement
+        self.domain = domain
+        self.symbol = symbol
+        self.prec = prec
+
+    @property
+    def ring(self) -> PowerSeriesRingProto[TElement, Er]:
+        """Return the base ring of the power series."""
+        return cast(
+            "PowerSeriesRingProto[TElement, Er]",
+            _power_series_ring(self.domain, self.prec),
+        )
+
+    @property
+    def one(self) -> PowerSeriesElement[Er]:
+        return self.ground_new(self.domain.one)
+
+    @property
+    def zero(self) -> PowerSeriesElement[Er]:
+        return self.ground_new(self.domain.zero)
```

Comment:
If the elements are immutable then this should be stored in the constructor as `self.one`.